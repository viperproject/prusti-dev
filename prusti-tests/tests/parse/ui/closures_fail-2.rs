// compile-flags: -Pprint_desugared_specs=true -Pprint_typeckd_specs=true -Pno_verify=true -Phide_uuids=true
// normalize-stdout-test: "[a-z0-9]{32}" -> "$(NUM_UUID)"
// normalize-stdout-test: "[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}" -> "$(UUID)"
// normalize-stdout-test: "\[[a-z0-9]{4}\]::" -> "[$(CRATE_ID)]::"

/// Failure case for closure macro parsing: missing return type.

use prusti_contracts::*;

fn main() {
    let f = closure!(
        requires(i >= 0),
        ensures(result == i + 1),
        |i: i32| i + 1
    );
}
