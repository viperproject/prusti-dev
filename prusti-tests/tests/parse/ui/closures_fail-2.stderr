error: closure must specify return type
  --> $DIR/closures_fail-2.rs:11:13
   |
11 |       let f = closure!(
   |  _____________^
12 | |         requires(i >= 0),
13 | |         ensures(result == i + 1),
14 | |         |i: i32| i + 1
15 | |     );
   | |_____^
   |
   = note: this error originates in the macro `closure` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to previous error

